cloudinit/net/__init__.py

```diff
@@ -765,6 +788,40 @@ def get_ib_interface_hwaddr(ifname, ethernet_format):
 
 
 def get_interfaces_by_mac():
+    if util.is_FreeBSD():
```

Comment:
I'd like to see this repeated pattern cleaned up to use separate classes for Linux and FreeBSD (with a common, documented interface/superclass). Or maybe it's indicating that these functions should actually move onto the `Renderer` class, with the FreeBSD implementation overriding the super-class's Linux implementation.

Not something that needs to happen in this PR at all, but would clean things up, I think.

@igalic This might be a good place to start with the investigation that we've discussed recently. Please make sure you and Goneri are on the same page, though, so we don't end up with duplicated effort!

Reply:
I agree this is not super elegant, but I also don't want to start a large refactoring in this patchset. @igalic's initiative goes in the good direction, especially, if NetBSD join the club.